services/etcd.go

```diff
@@ -672,3 +684,31 @@ func DownloadEtcdSnapshotFromBackupServer(ctx context.Context, etcdHost *hosts.H
 	}
 	return docker.RemoveContainer(ctx, etcdHost.DClient, etcdHost.Address, EtcdDownloadBackupContainerName)
 }
+
+func setEtcdPermissions(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, alpineImage string, process v3.Process) error {
+	var dataBind string
+
+	cmd := fmt.Sprintf("chmod 700 %s", EtcdDataDir)
+	if len(process.User) != 0 {
+		cmd = fmt.Sprintf("chmod 700 %s ; chown -R %s %s", EtcdDataDir, process.User, EtcdDataDir)
+	}
+	imageCfg := &container.Config{
+		Cmd: []string{
+			"sh", "-c",
+			cmd,
+		},
+		Image: alpineImage,
+	}
+	for _, bind := range process.Binds {
+		if strings.Contains(bind, "/var/lib/etcd") {
+			dataBind = bind
+		}
+	}
+	hostCfg := &container.HostConfig{
+		Binds: []string{dataBind},
+	}
+	if err := docker.DoRunContainer(ctx, etcdHost.DClient, imageCfg, hostCfg, EtcdPermFixContainerName, etcdHost.Address, ETCDRole, prsMap); err != nil {
```

Comment:
Can we start using `DoRunOnetimeContainer` for onetime containers? This mainly helps with waiting for the container to be done before proceeding, otherwise Rancher will continue before it exits.